ansari_db.py

```diff
@@ -160,10 +199,18 @@ def account_exists(self, email):
             if cur:
                 cur.close()
 
-    def save_token(self, user_id, token):
+    def save_token(self, user_id, token, token_type="login"):
         try:
+            if token_type == "login":
+                db_table = "user_tokens"
+            elif token_type == "refresh":
+                db_table = "refresh_tokens"
+            else:
+                raise ValueError(f"Unknown token type: {token_type}")
             cur = self.conn.cursor()
-            insert_cmd = "INSERT INTO user_tokens (user_id, token) " + \
+            # if I understand correctly, this means that the user can only login from one front-end
+            # at a time, logging in from another front-end would end his session on the first front-end.
+            insert_cmd = f"INSERT INTO {db_table} (user_id, token) " + \
```

Comment:
Yes, you are correct. This is a big. We should fix this bug and allow multiple tokens per user. Either make that fix here or add a TODO.

Reply:
Done.